backend/app/api.js

```diff
@@ -651,33 +651,99 @@ handlers.deleteAlbum = async function (info, req, res) {
  * }
  */
 handlers.renameAlbum = async function (info, req, res) {
-  if (false && !req.isAuthenticated()) {
+  if (!req.isAuthenticated()) {
     res.json({
       success: false,
       message: "You need to be logged in to rename an album",
     });
-  }
-  try {
-    const { albumId, newName } = info;
-    const success = await database.renameAlbum(albumId, newName);
-    if (success) {
+  } else {
+    try {
+      const { albumId, newName } = info;
+      const success = await database.renameAlbum(albumId, newName);
+      if (success) {
+        res.json({
+          success: true,
+          message: "Successfully renamed album",
+        });
+      } else {
+        res.json({
+          success: false,
+          message: "Failed to rename album due to unknown reason",
+        });
+      };
+    } catch (error) {
       res.json({
-        success: true,
-        message: "Successfully renamed album",
+        success: false,
+        message: error,
       });
-    } else {
+    }
+  };
+};
+// changes the ablums caption
+handlers.changeAlbumCaption = async function (info, req, res) {
+  if (!req.isAuthenticated()) {
+    res.json({
+      success: false,
+      message: "You need to be logged in to modify an album's caption",
+    });
+  } else {
+    try {
+      const { albumId, newCaption } = info;
+      const success = await database.changeAlbumCaption(albumId, newCaption);
+      if (success) {
+        res.json({
+          success: true,
+          message: "Successfully modified album's caption",
+        });
+      } else {
+        res.json({
+          success: false,
+          message: "Failed to modify album due to unknown reason",
+        });
+      };
+    } catch (error) {
       res.json({
         success: false,
-        message: "Failed to rename album due to unknown reason",

```

Comment:
Is the caption the same as the album's name? Otherwise it should say 'Failed to change album caption'

Reply:
No I think this is fine as the handler is to rename an album.